### PR TITLE
Fix 'versions' version to resolve #983

### LINKED ISSUE #983: Crash with non-standard ghc versions installed
Reproduce:

```hs
ghcup install ghc -u "https://gitlab.haskell.org/api/v4/projects/933/jobs/1746054/artifacts/ghc-x86_64-linux-fedora33-release.tar.xz" clc237
ghcup install ghc --force 8.10.7
[ Info  ] Removing the currently installed GHC version first!
[ Info  ] Removing ghc-x.y.z symlinks
[ Info  ] Removing/rewiring ghc-x.y symlinks
[ Info  ] Removing files safely from: /home/hasufell/.ghcup/ghc/8.10.7
ghcup: NonEmpty.fromList: empty list
CallStack (from HasCallStack):
  error, called at libraries/base/Data/List/NonEmpty.hs:200:15 in base:Data.List.NonEmpty
  fromList, called at ./Data/Versions.hs:907:30 in versions-6.0.4-4f1eb2a90b630d9ffb6129d75b6e171ea224af614d5012c00c2210b3e851a0c7:Data.Versions
```

This affects master and the released 0.1.20.0.

### PR DESCRIPTION
**Pull request by hasufell**
* https://github.com/haskell/ghcup-hs/issues/983
* https://github.com/fosskers/versions/pull/72

### CODE REVIEW

File: cabal.ghc948.project.freeze [hunk 1]
```diff
@@ -89,6 +89,7 @@ constraints: any.Cabal ==3.6.3.0 || ==3.8.1.0,
              any.dlist ==1.0,
              dlist -werror,
              any.exceptions ==0.10.5,
+             any.file-uri ==0.1.0.0,
```
Comment (runeksvendsen) [inline cabal.ghc948.project.freeze:92]: where did this come from?

File: cabal.ghc948.project [hunk 2]
```diff
@@ -12,6 +12,10 @@ else
 constraints: http-io-streams -brotli,
              any.aeson >= 2.0.1.0
 
+if os(mingw32)
+  if impl(ghc >= 9.4)
+    constraints: language-c >= 0.9.3
+
```
Comment (runeksvendsen) [inline cabal.ghc948.project:18]: where did this come from?

**Review by runeksvendsen — COMMENTED** [review]
I understand the change adding https://github.com/hasufell/versions.git@79e18fbc44fae3064d8957c550cc0229465db320 to cabal.project*.

But the `file-uri` change in .freeze files and the `constraints: language-c >= 0.9.3`-change seems unrelated to this PR.

File: cabal.ghc948.project (hunk 2 above)
Comment (hasufell) [reply cabal.ghc948.project:18]: Those changes come from the re-freeze, which were not consistently done after every single change to the project files.

**Review by runeksvendsen — APPROVED** [review]
I'd prefer to have the freeze-related changes as a separate commit/PR, but it's not important enough to block merging this.

**Pull request merged**